pkg/comp-functions/functions/common/non_sla_prom_rules.go

```diff
@@ -22,14 +22,29 @@ var (
 		"minio":      "minio",
 		"postgresql": "patroni",
 		"redis":      "redis",
+		"keycloak":   "keycloak",
 	}
+	minuteInterval, hourInterval, twoHourInterval promV1.Duration = "1m", "1h", "2h"
 )
 
-func GenerateNonSLAPromRules(obj client.Object) func(ctx context.Context, svc *runtime.ServiceRuntime) *fnproto.Result {
+type AlertsEnabled struct {
+	All           bool
+	DiskFillingUp bool
+	DiskSpace     bool
+	Memory        bool
+}
+
+type RuleStruct struct {
+	name, serviceName, instanceNamespace string
+	instanceNamespaceRegex               string
+	ruleList                             []promV1.Rule
+}
+
+func GenerateNonSLAPromRules(obj client.Object, alerts AlertsEnabled) func(ctx context.Context, svc *runtime.ServiceRuntime) *fnproto.Result {
```

Comment:
The initial idea was to use some sort of `RuleStruct` in here and not `AlertsEnabled`. From my point of view there is no need for 2 structs. We just need one and to expose it. My previous proposal was something around the definition of `RuleStruct` but we can also just use a set of booleans like you defined in `AlertsEnabled`. That's actually not bad. So I would remove RuleStruct all together and use here `AlertsEnabled`. Also I would remove the `All` `bool` from the struct and create a constructor called AlertsEnabledAll() that would set all the alert types to true. It's more clean that way. @Kidswiss WDYT?

Reply:
I'd still go with your first idea. Because it would generalize service specific metrics as well. With the current approach we only generalize the metrics provided by `kube-state-metrics`. However, it would be nice if this could be extended to deploy any other service specific alerts as well. For example max connections for PostgreSQL (which we dropped from the PnT composition but haven't moved to the comp functions, as I just noticed... ).